Sample.java

```diff
@@ -1,7 +1,97 @@
-// Time Complexity :
-// Space Complexity :
-// Did this code successfully run on Leetcode :
-// Any problem you faced while coding this :
+import java.util.*;
 
+class Sample {
+    class TreeNode {
+        int val;
+        TreeNode left;
+        TreeNode right;
+        TreeNode() {}
+        TreeNode(int val) { this.val = val; }
+        TreeNode(int val, TreeNode left, TreeNode right) {
+            this.val = val;
+            this.left = left;
+            this.right = right;
+        }
+    }
+    
+    // Time Complexity : O(N) N is number of noded

```

Comment:
N/2 leaf nodes, copying list from tempList into result for each leaf is O(N)=> So, O(N^2) worst case time complexity

Reply:
Yes, I didn't consider it while calculating. Thanks I updated the time complexity.